doc/source/whatsnew/v1.6.0.rst

```diff
@@ -35,6 +35,7 @@ Other enhancements
 - Added ``index`` parameter to :meth:`DataFrame.to_dict` (:issue:`46398`)
 - Added metadata propagation for binary operators on :class:`DataFrame` (:issue:`28283`)
 - :class:`.CategoricalConversionWarning`, :class:`.InvalidComparison`, :class:`.InvalidVersion`, :class:`.LossySetitemError`, and :class:`.NoBufferPresent` are now exposed in ``pandas.errors`` (:issue:`27656`)
+- Avoid showing deprecated signatures when introspecting functions that use ``deprecate_nonkeyword_arguments`` (:issue:`48692`)
```

Comment:
@MarcoGorelli Should this go in 1.5.1 if this PR is needed to (later) fix #48656 ?

Reply:
yeah I'd say so

Reply:
Probably in the "Other" section and then the PR that fixes #48656 in the "Regression" section